src/Response.php

```diff
@@ -122,7 +122,7 @@ public function withStatus($code, $reasonPhrase = '')
     {
         $new = clone $this;
         $new->statusCode = (int) $code;
-        if (!$reasonPhrase && isset(self::$phrases[$new->statusCode])) {
+        if ($reasonPhrase == '' && isset(self::$phrases[$new->statusCode])) {
```

Comment:
As quick way, and leaves to be permissive, it's more usual to use ````!$reasonPhrase```` than simple comparison about me.
Your simple condition is not better, and allow 0 in addition of null and '', as previously... whereas ````((null === $reasonPhrase) || ('' === $reasonPhrase))```` is verbose yes but consistent if you really want it.
About yoda condition, yes I know the code of this project and already met its use, I let you check.
That said, whatever do as you want.